bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ErrorValue.java

```diff
@@ -78,26 +79,49 @@ public ErrorValue(BType type, BString message, ErrorValue cause, Object details)
     @Override
     public String stringValue() {
         if (isEmptyDetail()) {
-            return "error " + getModuleName() + " (message=" + message.getValue() + ")";
+            return "error" + getModuleName() + "(" + ((StringValue) message).informalStringValue() + ")";
         }
-        return "error " + getModuleName() + " (message=" + message.getValue() + getCauseToString() + getDetailsToString() + ")";
+        return "error" + getModuleName() + "(" + ((StringValue) message).informalStringValue() + getCauseToString()
+                + getDetailsToString() + ")";
     }
 
     private String getCauseToString() {
         if (cause != null) {
-            return ", cause=" + ((BValue) cause).informalStringValue();
+            return ", " + ((BValue) cause).informalStringValue();
         }
         return "";
     }
 
     private String getDetailsToString() {
-        return ", detail=" + ((BValue) details).informalStringValue();
+        StringJoiner sj = new StringJoiner(", ");
+        for (Object key : ((MapValue) details).getKeys()) {
+            Object value = ((MapValue) details).get(key);
+            if (value == null) {
+                sj.add(key + "=null");
```

Comment:
should this be null,(), or space?

Reply:
it should be null. In the spec it says V1, V2 should be in the informal style
error D1&D2 (M, C, F1=V1,F2=V2)